### ISSUE TITLE
docker-sys interface not configured properly with Docker args

### ISSUE THREAD

**Issue by jbrockopp** [rancher/os#2478]
**RancherOS Version: (ros os version)**

```sh
[rancher@server ~]$ ros -v
version v1.4.0 from os image rancher/os:v1.4.0
```

**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**
OpenStack

This is potentially a bug and I'm opening this issue for clarification of expected behavior with the new way to configure the system-docker bridge network. I've already reviewed https://github.com/rancher/os/issues/2329 and https://github.com/rancher/os/issues/2422.

After following the instructions in the above issues, I've opened a new one because I believe the network isn't being properly configured. At least it's not being configured how I would expect it.

My cloud-config:

```yaml
#cloud-config
rancher:
  docker:
    extra_args:
      - "--bip=192.169.0.1/16"
  system_docker:
    extra_args:
      - "--bip=192.170.0.1/16"
```

The above cloud-config produces the user-docker network as I expect. Output below. 

```
[rancher@server ~]$ ifconfig docker0
docker0   Link encap:Ethernet  HWaddr 02:42:85:BE:5D:2E
          inet addr:192.169.0.1  Bcast:192.169.255.255  Mask:255.255.0.0
          ...
```

But it doesn't configure the system-docker network as I expect. Output below:

```
[rancher@server ~]$ ifconfig docker-sys
docker-sys Link encap:Ethernet  HWaddr 02:42:6F:2D:38:1C
          inet addr:192.170.0.1  Bcast:0.0.0.0  Mask:255.255.0.0
          ...
```

I would expect that the broadcast address (`Bcast`) for the `docker-sys` interface would be set just like it is for the `docker0` interface.

1. Is this expected behavior?
2. If not, are their workarounds? 
  * The only one we've come up with is running `ifconfig` on the server to properly configure the broadcast address for the interface.

**Comment by niusmallnan** [comment]
Our system-docker is based on docker 17.06. I guess this is a problem with docker itself.

I tried to switch user-docker to 17.06, then reboot the machine.
```
[root@ip-172-31-17-143 ~]# docker version
Server:
 Version:      17.06.2-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 19:59:19 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

I checked the docker0, it's the same with docker-sys. I cannot see the broadcast address:
```
[root@ip-172-31-17-143 ~]# ifconfig docker0
docker0   Link encap:Ethernet  HWaddr 02:42:6C:F2:61:BE
          inet addr:172.17.0.1  Bcast:0.0.0.0  Mask:255.255.0.0
          inet6 addr: fe80::42:6cff:fef2:61be/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:10 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:836 (836.0 B)
```

So this should be a problem with the early version of Docker, but it was fixed later.

I need backport some commits to our system-docker if we need to fix it.

BTW, what impact does this issue have on you?

**Comment by niusmallnan** [comment]
There may be a workaround, just add these to the cloud-config:
```
runcmd:
- ip addr del 172.18.42.1/16 dev docker-sys
- ip addr add 172.18.42.1/16 brd + dev docker-sys
```

Then reboot, check again:
```
[root@ip-172-31-17-143 rancher]# ifconfig docker-sys
docker-sys Link encap:Ethernet  HWaddr 02:42:3E:34:E0:29
          inet addr:172.18.42.1  Bcast:172.18.255.255  Mask:255.255.0.0
          inet6 addr: fe80::42:3eff:fe34:e029/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:7 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:606 (606.0 B)
```

**Comment by jbrockopp** [comment]
@niusmallnan thanks for getting back to us!

And thanks for providing those workarounds!

This currently hasn't caused use any issues.... but was noticed when troubleshooting the bip config changes.

If it is fixed in a future version of Docker then I think we can close this.

Do you happen to know what version of Docker this was fixed in?

**Comment by niusmallnan** [comment]
Fixed in docker 17.12.0-ce.
